### PR TITLE
Fetch profile response fixes: destroy

### PR DESCRIPTION
**Pull request by ImMPrada**
This PR allows FetchProfile to destroy an existing profile in DB, but not existing in github

### CODE REVIEW

File: app/services/sync/sync_profile.rb [hunk 1]
```diff
@@ -18,6 +18,11 @@ def update_profile(current_profile)
       profile
     end
 
+    def delete_profile(current_profile)
+      current_profile.destroy
+      self.profile = nil
+    end
+
```
Comment (mapra99) [inline app/services/sync/sync_profile.rb:25]: maybe worth adding a sec for this?

File: spec/services/sync/fetch_profile_spec.rb [hunk 2]
```diff
@@ -133,5 +133,41 @@
         expect(Profile.find_by(nickname: profile_name)[:url]).to eq(profile_url)
       end
     end
+
+    describe 'profile from Github: doesn\'t exist, profile from DB: exists' do
```
Comment (mapra99) [inline spec/services/sync/fetch_profile_spec.rb:137]: is this spec related to the additions of this PR?

File: app/services/sync/fetch_profile.rb [hunk 3]
```diff
@@ -29,6 +30,13 @@ def call_for_github_data
       self.code = profile_consumer.code
     end
 
+    def delete_profile
+      return if profile.blank? || code != 404
+
+      Sync::SyncProfile.new(github_profile).delete_profile(profile)
+      true
+    end
```
Comment (mapra99) [inline app/services/sync/fetch_profile.rb:38]: I think it's cleaner if you just do this on the method
```ruby
def delete_profile
  Sync::SyncProfile.new(github_profile).delete_profile(profile)
end
```

And in `#call` you do this:
```ruby
def call
  ...
  call_for_github_data
  return if profile.blank? && github_profile.blank?

  if code == 404
    delete_profile
    return 
   end
  ...
end
```

or also

```ruby
def call
  ...
  call_for_github_data
  return if profile.blank? && github_profile.blank?
  delete_profile and return if code == 404
  ...
end
```

Both options do the same.
Also `call_for_github_data` can be renamed to `fetch_github_data`, which seems to be what the method really does.

File: spec/services/sync/fetch_profile_spec.rb (hunk 2 above)
Comment (mapra99) [reply spec/services/sync/fetch_profile_spec.rb:137]: I understand now. You are adding a special case for `#call`, which would internally call `#delete_profile`

Comment (mapra99) [reply spec/services/sync/fetch_profile_spec.rb:137]: Also a nitpick
```suggestion
    describe "profile from Github: doesn't exist, profile from DB: exists" do
```

File: spec/services/sync/fetch_profile_spec.rb [hunk 4]
```diff
@@ -133,5 +133,41 @@
...
+          Github::ProfileConsumer,
+          call: {},
+          code: 404,
+          body: nil
+        )
+        allow(Github::ProfileConsumer).to receive(:new).and_return(profile_call_fetch_profile)
+
+        profile
+      end
+
```
Comment (mapra99) [inline spec/services/sync/fetch_profile_spec.rb:152]: In my opinion, it's a smell of how the spec is badly structured when you have to call the variable alone just for the sole purpose of loading the `let` declaration. I see 2 potential solutions here:
1. (Not recommented) Use `let!` instead of `let` to enforce the variable definition instead of lazy loading it.
2. For this case, do you really need to have a helper variable called `profile`? I see you are not using anywhere else in the spec, only here. So, why not just create the profile here on the `before` hook instead of inside the `let`? So it would look like this:
```suggestion
      let(:location)  { create(:location) }

      before do
        profile_call_fetch_profile = instance_double(
          Github::ProfileConsumer,
          call: {},
          code: 404,
          body: nil
        )
        allow(Github::ProfileConsumer).to receive(:new).and_return(profile_call_fetch_profile)

        create(:profile, nickname: profile_name, location: location)
      end
```

File: app/services/sync/fetch_profile.rb (hunk 3 above)
Comment (ImMPrada) [reply app/services/sync/fetch_profile.rb:38]:
```
if github_code == 404
  delete_profile
  return
end
```
`delete_profile && return if code == 404` seems not working, for example:

image

image

image

image

**Review by mapra99 — APPROVED** [review]
gog ogo go go
